arbnode/sequencer.go

```diff
@@ -506,3 +513,34 @@ func (s *Sequencer) Start(ctxIn context.Context) error {
 
 	return nil
 }
+
+func (s *Sequencer) StopAndWait() {
+	s.StopWaiter.StopAndWait()
+	if s.txRetryQueue.Len() == 0 && len(s.txQueue) == 0 {
+		return
+	}
+	// this usually means that coordinator's safe-shutdown-delay is too low
+	log.Warn("sequencer has queued items while shutting down", "txQueue", len(s.txQueue), "retryQueue", s.txRetryQueue.Len())
+	forwarder := s.GetForwarder()
+	if forwarder != nil {
+		for s.txRetryQueue.Len() > 0 {
+			item := s.txRetryQueue.Pop()
+			err := forwarder.PublishTransaction(item.ctx, item.tx)
+			if err != nil {
+				log.Warn("failed to forward transaction while shutting down", "source", "txRetryQueue", "err", err)
+			}
+		}
+	emptyqueue:
+		for {
+			select {
+			case item := <-s.txQueue:
+				err := forwarder.PublishTransaction(item.ctx, item.tx)
+				if err != nil {
+					log.Warn("failed to forward transaction while shutting down", "source", "txQueue", "err", err)
```

Comment:
This isn't necessary and shouldn't block merging, but could we merge the above loop into this one to deduplicate the PublishTransaction call and error code, like this pseudocode?
```
var item
if queue.Len() > 0 {
  item = queue.Pop()
} else {
  select ...
}
publish(item)
```